DOCSP-37533 Update & Replace Documents (#42)
mongodb/docs-pymongo: source/write/update.txt

Update Documents
================

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 2
   :class: singlecol

.. facet::
   :name: genre
   :values: reference

.. meta::
   :keywords: modify, change, bulk, code example

Overview
--------

In this guide, you can learn how to use {+driver-short+} to update
documents in a MongoDB collection by using the ``update_one()`` or
``update_many()`` methods.

.. .. tip:: Interactive Lab
   
..    This page includes a short interactive lab that demonstrates how to
..    modify data by using the ``update_many()`` method. You can complete this
..    lab directly in your browser window without installing MongoDB or a code editor.

..    To start the lab, click the :guilabel:`Open Interactive Tutorial` button at the
..    top of the page. To expand the lab to a full-screen format, click the
..    full-screen button (:guilabel:`⛶`) in the top-right corner of the lab pane.

Sample Data
~~~~~~~~~~~

The examples in this guide use the ``sample_restaurants.restaurants`` collection
from the :atlas:`Atlas sample datasets </sample-data>`. To learn how to create a
free MongoDB Atlas cluster and load the sample datasets, see the

> :ref:`<pymongo-get-started>` tutorial.

Reviewer: Broken link text

Update Operations
-----------------

> You can perform update operations in MongoDB with the following methods:

Reviewer: ```suggestion
You can perform update operations in MongoDB by using the following methods:
```

The following example uses the ``update_one()`` method to update the ``name`` value of a
document named "Bagels N Buns" in the ``restaurants`` collection:

.. literalinclude:: /includes/write/update.py
   :start-after: start-update-one
   :end-before: end-update-one
   :language: python
   :copyable:

Update Many Documents
~~~~~~~~~~~~~~~~~~~~~

The following example uses the ``update_many()`` method to update all documents
with a ``cuisine`` value of "Pizza". After the update, the documents have a
``cuisine`` value of "Pasta".

.. literalinclude:: /includes/write/update.py
   :start-after: start-update-many
   :end-before: end-update-many
   :language: python
   :copyable:

Customize the Update Operation
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

The ``update_one()`` and ``update_many()`` methods optionally accept additional
parameters, which represent options you can use to configure the update
operation. If you don't specify any additional options, the driver does not customize
the update operation.

.. list-table::
   :widths: 30 70
   :header-rows: 1

   * - Property
     - Description

   * - ``upsert``
     - | Specifies whether the update operation performs an upsert operation if no 

>          documents match the query filter.

Reviewer: S: Link to upsert server docs

       | Defaults to ``False``

   * - ``bypass_document_validation``
     - | Specifies whether the update operation bypasses document validation. This lets you 
         update documents that don't meet the schema validation requirements, if any 
         exist. For more information about schema validation, see :manual:`Schema

>          Validation </core/schema-validation/#schema-validation>` on the MongoDB

Reviewer: ```suggestion
         Validation </core/schema-validation/#schema-validation>` in the MongoDB
```

         Server manual.
       | Defaults to ``False``.

   * - ``collation``
     - | Specifies the kind of language collation to use when sorting
         results. For more information, see :manual:`Collation </reference/collation/#std-label-collation>`

>          on the MongoDB Server manual.

Reviewer: ```suggestion
         in the MongoDB Server manual.
```

   * - ``array_filters``
     - | A list of filters that specifies which array elements an update applies
         to.

   * - ``hint``

>      - | Gets or sets the index to use to scan for documents. 

Reviewer: ```suggestion
     - | Gets or sets the index to scan for documents. 
```

         For more information, see the :manual:`hint statement </reference/command/update/#std-label-update-command-hint>`

>          on the MongoDB Server manual.

Reviewer: ```suggestion
         in the MongoDB Server manual.
```
